src/Lasso.jl

```diff
@@ -301,10 +322,191 @@ function StatsBase.fit{T<:AbstractFloat,V<:FPVector}(::Type{LassoPath},
+dispersion_parameter(path::RegularizationPath) = typeof(distfun(path)) in (Gamma, Normal, InverseGaussian)
+
+function StatsBase.loglikelihood(path::RegularizationPath)
+    n = nobs(path)
+    if typeof(path.m) <: LinearModel
+        -0.5*n.*log(deviance(path))
+    else
+        -0.5*n.*deviance(path)
+    end
+end
+
+if Pkg.installed("StatsBase") >= v"0.8.0"
+    import StatsBase.df, StatsBase.aicc
+end
+
+""" Approximates the degrees-of-freedom in each segment of the path as the number of non zero coefficients
+    plus a dispersion parameter when appropriate.
+    Note that for GammaLassoPath this may be a crude approximation, as gamlr does this differently.
+"""
+function df(path::RegularizationPath)
+    nλ = length(path.λ)
+    βs = coef(path)
+    dof = zeros(Int,nλ)
+    for s=1:nλ
+        dof[s] = sum(βs[:,s].!=0)
+    end
+
+    if dispersion_parameter(path)
+        # add one for dispersion_parameter
+        dof+=1
+    end
+
+    dof
+end
+
+function aicc(path::RegularizationPath;k=2)
+    d = df(path)
+    n = nobs(path)
+    ic = -2loglikelihood(path) + k*d + k*d.*(d+1)./(n-d-1)
+    ic[d.+1 .> n] = realmax(eltype(ic))
+    ic
+end
+
+minAICc(path::RegularizationPath;k=2)=indmin(aicc(path;k=k))
+
+hasintercept(path::RegularizationPath) = hasintercept(path.m.pp)
+
+#Consistent with StatsBase.coef, if the model has an intercept it is included.
+function StatsBase.coef(path::RegularizationPath; select=:all, nCVfolds=10)
+    if length(path.λ) == 0
+        X = path.m.pp.X
+        p = size(X,2)
+        if hasintercept(path)
+            p+=1
+        end
+        return zeros(eltype(X),p)
+    end
+
+    if select == :all
+        if hasintercept(path)
+            vcat(path.b0',path.coefs)
+        else
+            path.coefs
+        end
+    elseif select == :AICc
+        if hasintercept(path)
+            vec(vcat(path.b0[minAICc(path)],path.coefs[:,minAICc(path)]))
+        else
+            path.coefs[:,minAICc(path)]
+        end
+    elseif select == :CVmin || select == :CV1se
+        gen = Kfold(length(path.m.rr.y),nCVfolds)
+        segCV = cross_validate_path(path;gen=gen,select=select)
+        if hasintercept(path)
+            vec(vcat(path.b0[segCV],path.coefs[:,segCV]))
+        else
+            path.coefs[:,segCV]
+        end
+    else
+        error("unknown selector $select")
+    end
+end
+
+# function intercept(path::RegularizationPath; select=:all)
+#     if select == :all
+#         path.b0
+#     elseif select == :AICc
+#         path.b0[minAICc(path)]
+#     end
+# end
+
+## Prediction function for GLMs
+function StatsBase.predict{T<:AbstractFloat}(path::RegularizationPath, newX::AbstractMatrix{T}; offset::FPVector=Array(T,0), select=:all)
+    # add an interecept to newX if the model has one
+    if hasintercept(path)
+        newX = [ones(eltype(newX),size(newX,1),1) newX]
+    end
+
+    # calculate etas for each obs x segment
+    eta = newX * coef(path;select=select)
```

Comment:
Instead of constructing a copy of `newX` augmented by a column of ones and then multiplying it by a copy of the `path.coef` augmented by a row of zeros, it would be more efficient to simply compute `newX * path.coef + path.b0`.